form/src/main/java/org/dhis2/form/data/EnrollmentRepository.kt

```diff
@@ -1,53 +1,57 @@
-package org.dhis2.data.forms.dataentry
+package org.dhis2.form.data
 
 import io.reactivex.Flowable
 import io.reactivex.Single
-import java.util.ArrayList
 import org.dhis2.Bindings.userFriendlyValue
-import org.dhis2.data.dhislogic.DhisEnrollmentUtils
-import org.dhis2.form.data.DataEntryBaseRepository
+import org.dhis2.commons.date.DateUtils
+import org.dhis2.form.model.EnrollmentMode
 import org.dhis2.form.model.FieldUiModel
 import org.dhis2.form.model.SectionUiModelImpl.Companion.SINGLE_SECTION_UID
 import org.dhis2.form.ui.FieldViewModelFactory
-import org.dhis2.usescases.enrollment.EnrollmentActivity
-import org.dhis2.utils.DateUtils
-import org.dhis2.utils.DhisTextUtils
+import org.dhis2.form.ui.provider.EnrollmentFormLabelsProvider
 import org.hisp.dhis.android.core.D2
 import org.hisp.dhis.android.core.arch.helpers.UidsHelper.getUidsList
 import org.hisp.dhis.android.core.arch.repositories.scope.RepositoryScope
 import org.hisp.dhis.android.core.common.FeatureType
 import org.hisp.dhis.android.core.common.ObjectStyle
 import org.hisp.dhis.android.core.common.ValueType
 import org.hisp.dhis.android.core.enrollment.EnrollmentObjectRepository
+import org.hisp.dhis.android.core.imports.TrackerImportConflict
 import org.hisp.dhis.android.core.option.Option
 import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
 import org.hisp.dhis.android.core.program.Program
 import org.hisp.dhis.android.core.program.ProgramSection
 import org.hisp.dhis.android.core.program.ProgramTrackedEntityAttribute
 import org.hisp.dhis.android.core.program.SectionRenderingType
 import org.hisp.dhis.android.core.trackedentity.TrackedEntityAttribute
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityAttributeValueObjectRepository
 import timber.log.Timber
 
 class EnrollmentRepository(
     private val fieldFactory: FieldViewModelFactory,
     private val enrollmentUid: String,
     private val d2: D2,
-    private val dhisEnrollmentUtils: DhisEnrollmentUtils,
-    private val enrollmentMode: EnrollmentActivity.EnrollmentMode,
-    private val enrollmentDataSectionLabel: String,
-    private val singleSectionLabel: String,
-    private val enrollmentOrgUnitLabel: String,
-    private val teiCoordinatesLabel: String,
-    private val enrollmentCoordinatesLabel: String,
-    private val reservedValuesWarning: String,
-    private val enrollmentDateDefaultLabel: String,
-    private val incidentDateDefaultLabel: String
+    private val enrollmentMode: EnrollmentMode,
+    private val enrollmentFormLabelsProvider: EnrollmentFormLabelsProvider
 ) : DataEntryBaseRepository(d2, fieldFactory) {
 
     private val enrollmentRepository: EnrollmentObjectRepository =
         d2.enrollmentModule().enrollments().uid(enrollmentUid)
 
-    private val canEditAttributes: Boolean = dhisEnrollmentUtils.canBeEdited(enrollmentUid)
+    private val canEditAttributes: Boolean = canBeEdited()
+
```

Comment:
Is this method necessary? can we just called `canBeEdited()` or is there a reson to have it like this?